src/composables/queries/usePoolQuery.ts

```diff
@@ -100,13 +100,16 @@ export default function usePoolQuery(
       );
 
       linearPoolTokens.push(
-        ...linearPool.tokens.filter(
-          token => token.address !== linearPool.address
-        )
+        ...linearPool.tokens
+          .filter(token => token.address !== linearPool.address)
+          .map(token => ({ ...token, address: getAddress(token.address) }))
       );
     });
 
     pool.linearPoolTokens = linearPoolTokens;
+    pool.linearPoolTokensAddresses = linearPoolTokens.map(
+      token => token.address
+    );
```

Comment:
I'm starting to think a single `linearPoolTokenMap` might be better here?

Reply:
Agree, this would allow for an easy `Object.keys` / `Object.values` for both scenarios. Fixed that.